src/main/java/org/commcare/cases/query/QuerySensitiveTreeElement.java

```diff
@@ -0,0 +1,36 @@
+package org.commcare.cases.query;
+
+import org.javarosa.core.model.instance.AbstractTreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+
+/**
+ * NOTE: Each time a method is implemented here, it should be added to QuerySensitiveTreeElementWrapper
+ *
+ * Created by ctsims on 9/19/2017.
+ */
+
+public interface QuerySensitiveTreeElement<T extends AbstractTreeElement> extends AbstractTreeElement<T> {
+
+    /**
+     * Retrieves the TreeElement representing the attribute at
+     * the provided namespace and name, or null if none exists.
+     *
+     * If 'null' is provided for the namespace, it will match the first
+     * attribute with the matching name.
+     */
+    T getAttribute(QueryContext context, String namespace, String name);
+
+    int getChildMultiplicity(QueryContext context, String name);
+
+    /**
+     * Get a child element with the given name and occurence position (multiplicity)
+     *
+     * @param name         the name of the child element to select
+     * @param multiplicity is the n-th occurence of an element with a given name
+     */
+    T getChild(QueryContext context, String name, int multiplicity);
+
+    boolean hasChildren(QueryContext context);
```

Comment:
just consistency of not having to change the signature of this interface in the event that a future implementation would require the context to be around. It makes the most sense to me that implementing classes provide all the hooks up front, rather than pick and choose, which seems a bit error prone to me.

Reply:
Hm, I don't think I agree with this. If this method isn't being used, it's adding the appearance of additional complexity that isn't there (to something that already appears really complex). I also don't see how leaving it out creates any more room for error.

Reply:
Ah, I understand. Sorry, I was thinking about the wrapper.

You're right, this didn't make sense. I removed them and also went through and made sure everything was tagged with @Override to make sure that we wouldn't miss implementing/de-implementing